CRM/Contribute/Form/Search.php

```diff
@@ -390,6 +390,11 @@ public function fixFormValues() {
       $this->_defaults['contribution_status_id'] = array($status => 1);
     }
 
+    $pcpid=CRM_Utils_Request::retrieve('pcpid', 'Positive', $this);
+    if ($pcpid) {
```

Comment:
Thanks for your explanation, i am focusing on this also, and your last commit is correct, but, in term from the FORM, users can search using multiples parameters, we should to permit this in method, it could to function with string, array or others method, that's all.

Repeating again, your last commit is correct and good, you are taking my point of view, i think it.

Thanks :+1: